File: 3393-make-string-anti-palindrome/make-string-anti-palindrome.py
class Solution:
    def makeAntiPalindrome(self, s: str) -> str:        
        n = len(s)
        ans = sorted(ch for ch in s)
        i=j=n//2 
        while j<n and ans[j]==ans[i]:
            j += 1
        while ans[i]==ans[n-1-i]:
            if j==n: return "-1"
            ans[i],ans[j]=ans[j],ans[i]
            i+=1
            j+=1
        
        return "".join(ans)

    

File: 3393-make-string-anti-palindrome/make-string-anti-palindrome.java
class Solution {
    public String makeAntiPalindrome(String s) {
        char[] ans = s.toCharArray();
        Arrays.sort(ans);

        int n = ans.length;
        int j = n/2;
        for (int i=n/2-1;i>=0;i--) {
            if (ans[i]!=ans[n-1-i])
                break;
            // swap n-1-i with j
            while (j<n && ans[j]==ans[n-1-i]) {
                j++;
            }
            if (j>=n) return "-1";
            // swap 
            char tmp = ans[j];
            ans[j] = ans[n-1-i];
            ans[n-1-i] = tmp;
            j += 1;
        }

        return new String(ans);
    }
}